8.0 port product standard price vat incl

legalsylvain opened the pull request:
I propose in OCA a module that add a standard_price_vat_incl field, based on standard_price and taxes_id fields.
This is a port of a custom module in V7 (https://github.com/grap/odoo-addons-misc/tree/7.0).

This module is usefull in the below setting  
- You set your taxes with VAT included;
- You want to have pricelist based on standard price ; (standard_price + 10%)

without this module, the computed price will be bad, because it will not take into account the vat, so you will sale : standard_price + 10% - VAT ratio. 

Review welcome.

legalsylvain committed:
[WIP] port the module 'product_standard_price_vat_incl' from V7  ;

legalsylvain committed:
[REF] PORT in V8 product_standard_price_vat_incl : use new API and convention;

legalsylvain committed:
[REF] tests are now using V8 API;

legalsylvain committed:
[REF] rename tests file;

moylop260 commented:
@legalsylvain 
The module is fine!
Thanks for you efforts

Do you have considered that exists `tax included` in `account.tax` model?
![tax_included](https://cloud.githubusercontent.com/assets/6644187/9800401/5fbc7576-57d2-11e5-909a-7923046528a0.png)

legalsylvain commented:
hi @moylop260. Thanks for your review. Yes, I know VAT Incl checkbox, because, this module fix a "bug", if we check this checkbox.

Let we take a simple example : 
We have a product with 
- standard_price (ie : standard_price vat excl) = 10€ 
- Taxes_id = 20% **VAT INCL**;
- list_price = 24€ ; (so, VAT incl); ie sale price = 20€ VAT excl

OK.
so, if we set a pricelist, based on standard price + 50%, we expect to have a sale price of 10 + 50% = 15 € vat excl = 18 vat incl.
without this module, Odoo will compute a sale price of 15 VAT Incl, that is false.

This module allow user to have pricelist based on standard_price, if taxes are VAT Incl.

Kind regard.

moylop260 commented:
@legalsylvain 
Thank for explanation
:+1: 

Note: Do you have a reference to a recent issue reported in github?

legalsylvain commented:
Hi again, @moylop260. IMO, it's not a Odoo bug. when we use taxes with vat incl, we assume that price are vat incl. (so, price base on standard_price that are vat excl will not work by design).

moylop260 commented:
I got it
Thank you

pedrobaeza commented:
I think you're not making the calculations the correct way. If you have an untaxed price of 10 € and you put 20% vat included, the price you have to have in the product is not 10 €, but 12 € (10 + 1.2). If you add a 50% in this price, the final price is the 18 € that you're looking for.

legalsylvain commented:
Hi @pedrobaeza. Thanks a lot for taking time to review my use case.
For me :
1. the 'list_price' can be vat excl or var incl depending of taxes configuration. (vat incl used by B2C companies, and vat excl used by B2B companies);
2. the 'standard_price' is for me **allways** vat exc : because :  
   - stock valuation are allways vat exc, and stock valuation are based on standard_price ; 
   - by definition, a Cost is vat exc ; (it is what my users say);
   - This kind of module [product_standard_margin](https://github.com/OCA/margin-analysis/blob/8.0/product_standard_margin/models/product_product.py#L43) should be fixed, if my PoV is wrong. (with replenishment_cost = standard_price [Ref](https://github.com/OCA/margin-analysis/blob/8.0/product_replenishment_cost/models/product_product.py#L34) )

Did I missed something about that point ?

But even if I missed something, and it's possible to have standard_price vat incl, it is not my setting.
- All my sale vat are VAT INCL;
- All my purchase vat are VAT EXCL;
  That is for me a valid configuration, and this module allow to use pricelist based on cost (standard_price) in this configuration.

legalsylvain commented:
@pedrobaeza : did you have the time to review my last comment ?
kind regards.

pedrobaeza commented on product_standard_price_vat_incl/README.rst:
```diff
@@ -0,0 +1,48 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Brings a standard Price with VAT Included
```
Put upper line

pedrobaeza commented on product_standard_price_vat_incl/README.rst:
```diff
@@ -0,0 +1,48 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
```
Put target line

pedrobaeza commented on product_standard_price_vat_incl/README.rst:
```diff
@@ -0,0 +1,48 @@
+will be bad, because the VAT will be not correctly computed;
+
+
+For further information, please visit:
```
Include Try me on runbot button

pedrobaeza commented on product_standard_price_vat_incl/README.rst:
```diff
@@ -0,0 +1,48 @@
+=========================================
+
+This module is interesting in the following case:
+    * You use VAT With Price Included;
```
s/VAT With Price/Price with VAT

pedrobaeza commented:
OK, now I understand your use case, although it's a bit special.

Please add a configuration section where you explain that a new price type is added selectable on pricelist called...

The rest looks OK.

pedrobaeza commented on product_standard_price_vat_incl/__init__.py:
```diff
@@ -0,0 +1,2 @@
+# -*- encoding: utf-8 -*-
```
Please replace all occurrences of encoding with coding

legalsylvain committed:
[REF] New OCA Convention : change 'encoding' into 'coding';

legalsylvain committed:
[REF] product_standard_price_tax_included : change name of the module, replacing 'VAT' by 'Tax' ;
[IMP] readme description  and add screenshot;
[ADD] pot file;

legalsylvain committed:
[FIX] icon

legalsylvain committed:
[IMP] description and images;

legalsylvain commented:
Hi @pedrobaeza. I refactored a bit this module, renaming it, improving description adding screenshot...
Please let me know if it is more understandable now :
https://github.com/grap/product-attribute/blob/8.0_PORT_product_standard_price_vat_incl/product_standard_price_tax_included/README.rst
I also added runbot links ...

dcorio commented:
:-1: 
1) if you don't enable work activities (Settings > Project > Log work activities on tasks) you get an empty and useless Description tab.
2) have both Briefing and Description tabs can be quite misleading

to achieve what you propose, i would suggest to move the work activities (work_ids) to a dedicated tab.

legalsylvain commented:
@dcorio : I didn't understood your comment. I guess you let a comment in the wrong PR. Could you check please, and delete / update your comment ? 
(My PR doesn't have activities or work_ids defined).
Kind regards.

legalsylvain committed:
[FIX] use new API;

legalsylvain commented:
Hi @pedrobaeza. Is it now ok for you ? kind regards.

pedrobaeza commented on product_standard_price_tax_included/README.rst:
```diff
@@ -0,0 +1,75 @@
+So without this module, the sale price will be bad in this below configuration:
+
+* Price list based on the field 'Cost Price';
+  (If you want a fixed margin, for exemple: Cost Price + 10%);
```
s/exemple/example

pedrobaeza commented on product_standard_price_tax_included/README.rst:
```diff
@@ -0,0 +1,75 @@
+=======================================================
+
+In Odoo, the Cost Price Field (standard_price) is by definition 'Tax Excluded'.
+So without this module, the sale price will be bad in this below configuration:
```
s/in this below configuration/in the following configuration

pedrobaeza commented on product_standard_price_tax_included/README.rst:
```diff
@@ -0,0 +1,75 @@
+  (If you want a fixed margin, for exemple: Cost Price + 10%);
+* Products set with Sale Taxes "Tax Included" ; (B2C settings)
+
+This module fix the problem, adding a new field 'Cost Price Tax Included'
```
s/fix/fixes

pedrobaeza commented on product_standard_price_tax_included/README.rst:
```diff
@@ -0,0 +1,75 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/product-attribute/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/product-attribute/issues/new?body=module:%20product_standard_price_tax_included%0Aversion:%208.0.1.0.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Put version 8.0 (Odoo version), not the module version

pedrobaeza commented:
Little remarks, but overall :+1:

legalsylvain committed:
[FIX] typo in readme file;

legalsylvain commented:
@pedrobaeza. I took into account your remarks. (with a big delay, sorry, I forgot this PR !)
I think it's now mergeable.

regards.

pedrobaeza commented:
:+1:
